### PR TITLE
Feedback and Customer Usage Improvements

### PR DESCRIPTION
**Pull request by vanmaegima**
## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

Improve Private Registries examples and fioctl configure-git usage.

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for f6ae42fbcf0e24dfa44d27d4e9d7b20f14940485 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1926/docs/artifacts/html/index.html

File: source/getting-started/git-config/index.rst [hunk 1]
```diff
@@ -6,38 +6,6 @@ Configuring Git
 Pushing to your repositories with :ref:`FoundriesFactory® <ref-factory-definition>` is a matter of configuring Git to use :ref:`Fioctl™ <ref-fioctl>` as a credential helper.
 Afterwards, Git will know when you are connecting to ``source.foundries.io`` and will use :ref:`ref-fioctl` for authentication when utilizing ``git`` commands.
 
-Source Code Access Token
```
Comment (doanac) [inline source/getting-started/git-config/index.rst:9]: Maybe we keep this content, but bury it somewhere in the reference manual. Seems like some day, somebody will find a reason for needing it.

File: source/reference-manual/docker/private-registries.rst [hunk 2]
```diff
@@ -67,44 +107,10 @@ credentials_.
...
- # containers.git <CONTAINER>/Dockerfile
- FROM <YOUR REGISTRY>.azurecr.io
-
-Then compose apps can reference this hub.foundries.io container image.
-
-.. _securely access:
-   https://docs.microsoft.com/en-us/answers/questions/734990/iot-device-authentication-with-acr.html
+Another suggested approach is to follow the procedure as outlined for devices
+accessing the :ref:`Azure Container Registry <ref-acr-devices>`. This wraps the
+container images into hub.foundries.io and with this setup, devices rely on the
```
Comment (doanac) [inline source/reference-manual/docker/private-registries.rst:112]: I'm not sure I'd call this a "suggested" approach. Maybe just "Another approach".

File: source/getting-started/git-config/index.rst [hunk 3]
```diff
@@ -26,6 +26,10 @@ Run the following command to add the relevant entries to the Git configuration:
       git config --global credential.https://source.foundries.io.helper fio
       ln -s /usr/bin/fioctl /usr/bin/git-credential-fio
 
+   Existing users may need to remove the following lines from ``.gitconfig`` to use ``fioctl configure-git`` utility::
+
+      [http "https://source.foundries.io"]
+      extraheader = Authorization: basic <TOKEN>
```
Comment (doanac) [inline source/getting-started/git-config/index.rst:32]: Should we also mention that you need to run `fiocl login --refresh-access-token` when you change your oauth scopes?

File: source/reference-manual/docker/private-registries.rst [hunk 4]
```diff
@@ -20,6 +20,46 @@ in order to construct a Target. The :ref:`ref-factory-definition`
 (``ci-scripts.git``) provides a way to configure this.
 
 
+Configuring for CI Azure Container Registry(ACR)
```
Comment (kprosise) [inline source/reference-manual/docker/private-registries.rst:23]:
```suggestion
Configuring for CI Azure Container Registry (ACR)
```

File: source/reference-manual/docker/private-registries.rst [hunk 5]
```diff
@@ -20,6 +20,46 @@ in order to construct a Target. The :ref:`ref-factory-definition`
...
+------------------------------------------------
```
Comment (kprosise) [inline source/reference-manual/docker/private-registries.rst:24]:
```suggestion
-------------------------------------------------
```

File: source/reference-manual/docker/private-registries.rst (hunk 2 above)
Comment (kprosise) [inline source/reference-manual/docker/private-registries.rst:112]:
```suggestion
container images into `hub.foundries.io`. With this setup, devices rely on the
```

File: source/reference-manual/docker/private-registries.rst [hunk 6]
```diff
@@ -20,6 +20,46 @@ in order to construct a Target. The :ref:`ref-factory-definition`
...
+.. _ref-acr-devices:
+
+Configuring Devices for ACR
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Azure does not include a built-in way for devices to `securely access`_
+a container registry like AWS IoT does. The recommended approach for
+this scenario is pushing container images from ACR into
+hub.foundries.io or having a Foundries container image built with
+something like::
```
Comment (kprosise) [inline source/reference-manual/docker/private-registries.rst:53]:
```suggestion
something similar to::
```

**Review by kprosise — APPROVED** [review]
Suggestions made.

**Comment by doanac** [comment]
Docs for c45ed9bd9bddbcdd8994d08b3c9e72ad5ccabe57 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1934/docs/artifacts/html/index.html

**Comment by vanmaegima** [comment]
@kprosise can you please review the vale error and check if I can ignore these? Errors seem to be incorrectly complaining about the acronyms and hub.foundries.io.

**Comment by doanac** [comment]
Docs for f23b0e3c3dddbdc5927b02ebebfcc5cb7c458ffa are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1935/docs/artifacts/html/index.html

File: source/reference-manual/docker/private-registries.rst [hunk 7]
```diff
@@ -20,6 +20,46 @@ in order to construct a Target. The :ref:`ref-factory-definition`
...
+
+.. _ref-acr-devices:
+
+Configuring Devices for ACR
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Azure does not include a built-in way for devices to `securely access`_
+a container registry like AWS IoT does. The recommended approach for
+this scenario is pushing container images from ACR into
+`hub.foundries.io` or having a Foundries container image built with
```
Comment (kprosise) [inline source/reference-manual/docker/private-registries.rst:52]:
```suggestion
`hub.foundries.io` or having a FoundriesFactory container image built with
```

**Comment by kprosise** [comment]
> @kprosise can you please review the vale error and check if I can ignore these? Errors seem to be incorrectly complaining about the acronyms and hub.foundries.io.

Yeah. The ACR acronym may be popping up as it is defined in a header rather than the body. As for flagging the "foundries.io" in hubs.foundries.io, that is indeed odd as it shouldn't be triggered by anything in literals. I will take a look and see what's going on there, but feel free to move forward with this PR.

**Comment by doanac** [comment]
Docs for 1d0a7017911d92da309b50ca15f271ba1b681fc9 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1938/docs/artifacts/html/index.html

**Pull request merged**